code/game/gamemodes/game_mode.dm

```diff
@@ -81,6 +83,46 @@
 /datum/game_mode/proc/pre_setup()
 	return 1
 
+/datum/game_mode/proc/create_special_antags()
+	var/list/living_crew = list()
+	living_crew = get_living_crew()
+
+	var/list/candidates = list()
+	for(var/mob/living/carbon/human/H in living_crew)
+		if((!H.client) || (is_centcom_level(H.z)))
+			continue
+		if(is_banned_from(H.ckey, list(ROLE_SPECIAL)))
+			continue
+		candidates += H
+
+	for(var/role_to_init in allowed_special)
+		var/datum/special_role/new_role = new role_to_init
+		if(!prob(new_role.probability))
+			continue
+		active_specials += new_role
+
+	for(var/datum/special_role/special in active_specials)
+		if(special.spawn_mode == "midround")
+			continue
+		//To make it feel a little more random, and for efficiency reasons we just pick the person, then check their job and if they cannot be antag, we will just remove the slot
+		var/amount = round(living_crew.len * special.proportion)
+		amount = min(amount, special.max_amount)
+		for(var/i in 1 to amount)
+			var/mob/person = pick(candidates)

```

Comment:
Further evidence that your procs should be returning mobs instead of minds.

Reply:
Also you want `pick_n_take(candidates)` then you don't have to worry about removing them from the list.